### ISSUE TITLE
File Picker redirects to website instead of closing and calling the pickers success function

### ISSUE THREAD

**Issue by cs-NET** [OneDrive/onedrive-api-docs#215]
After selecting a file and clicking open the Picker redirects to my website within the picker popup window instead of closing the picker popup and calling the success function.

The only unknown (part missing from the documentation) is what the exact Application settings should be when creating the application / client id. I was thinking maybe the **Mobile or desktop client app** setting was at fault but changing that does not seem to resolve the issue.

What would cause the picker to redirect instead of closing?

**Comment by jscaria** [comment]
Can you provide me with more details?   Did this work for you previously?   Is the redirect URL for your client ID a URL that's referencing the SDK?

**Comment by cs-NET** [comment]
I have resolved the issue after understanding the picker workflow a bit better. 

The problem was that I was lazy loading the onedrive.js javascript file when the user navigated to the portion of my single page app that opens the onedrive picker. The picker would open fine however on the redirect back the onedrive.js file was not loaded. This results in the website remaining loaded in the picker window. I assume that on the redirect back there must be something in the onedrive.js script that parses the query string that results in calling the success callback on the originating page.

**TL;DR** I had to remove the lazy loading and hard code the onedrive.js script into my SPA.

**Comment by jscaria** [comment]
Yes, you're correct.    Under the covers, the popup window actually navigates to the redirect URI with details about the selected file(s) that the parent window is looking for.    

Since this issue has been resolved, I'm closing it out.

**Comment by hemantsalunke90** [comment]
I am developing an hybrid app. I want to have a functionality like [https://dev.onedrive.com/sdk/js-v7/js-picker-open.htm] on my device but If I followed the same code, I am getting an error after clicking on open. The error is like 
![image](https://cloud.githubusercontent.com/assets/20579097/17372392/4c60d99e-59c1-11e6-823b-42b59d0161ab.png)
